src/hu_stuff.c

```diff
@@ -1528,7 +1528,8 @@ void HU_Erase(void)
   w = doom_widget;
   while (w->multiline)
   {
-    if (*w->multiline->on)
+    if (*w->multiline->on ||
```

Comment:
Hm, I have to admit I don't quite get it. `w_message.on` points to `message_on`. Why can't we rely on its value to decide if the message widget needs to be cleared or not?

Reply:
`w_message.on` disabled, so it is not erased and not drawn over, it just stays in the background. I don't know why, the message logic is confusing.

Reply:
Does it affect chat messages as well? And the secret revealed message if the screen size is reduced enough to have it drawn into the border? I'm asking because Vanilla MBF erases all Doom widgets unconditionally:

https://github.com/doomtech/winmbf/blob/300f3516f6dc8dfab057989a141db84585efd924/Source/hu_stuff.c#L1195

Reply:
It seems chat mesages works fine.

> I'm asking because Vanilla MBF erases all Doom widgets unconditionally

It works too, but we need to improve the logic of `HUlib_erase_widget` - don't erase the whole `SCREENWIDTH` as it looks bad with some widgets.

Reply:
Yes, it needs some logic to only erase the left and right frame around the view window.

Reply:
Done here: https://github.com/fabiangreffrath/woof/pull/1478/commits/198c57f478b13694b64ff83c87f32248eccac921

Reply:
Nope, sorry, this still doesn't erase lines which cross the view window.

![woof0000](https://github.com/fabiangreffrath/woof/assets/6449237/fd81b2de-0b52-401f-9173-1a6dc34deec9)
![woof0001](https://github.com/fabiangreffrath/woof/assets/6449237/4fcfe3fe-8fee-43a2-8619-21efe4bd3f93)

This is with the following `woofhud.lmp`

```
hud 1
message 0 6
time bottomright
coords topleft
stats topright
```

Reply:
Naive patch (I'm tired):

```patch
--- a/src/hu_lib.c
+++ b/src/hu_lib.c
@@ -510,17 +510,25 @@ void HUlib_erase_widget (const hu_widget_t *const w)
   const hu_multiline_t *const m = w->multiline;
   const hu_font_t *const f = *m->font;
 
-  const int height = m->numlines * f->line_height;
-  const int y = vert_align_widget(w, m, f, w->h_align, w->v_align);
+  int height = m->numlines * f->line_height;
+  int y = vert_align_widget(w, m, f, w->h_align, w->v_align);
 
-  if (y + height > scaledviewy && y < scaledviewy + scaledviewheight - height)
+  if (y + height < scaledviewy || y >= scaledviewy + scaledviewheight)
   {
-    R_VideoErase(0, y, scaledviewx, height);
-    R_VideoErase(scaledviewx + scaledviewwidth, y, scaledviewx, height);
+    R_VideoErase(0, y, video.unscaledw, height);
   }
   else
   {
-    R_VideoErase(0, y, video.unscaledw, height);
+    while (height-- && y++)
+    {
+      if (y < scaledviewy || y >= scaledviewy + scaledviewheight)
+        R_VideoErase(0, y, video.unscaledw, 1);
+      else
+      {
+        R_VideoErase(0, y, scaledviewx, 1);
+        R_VideoErase(scaledviewx + scaledviewwidth, y, scaledviewx, 1);
+      }
+    }
   }
 }
 
```

Reply:
I think problem was in border size: https://github.com/fabiangreffrath/woof/pull/1478/commits/bebcefbf55ed373ad287c1a863704a08af1409f9